compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -381,20 +392,47 @@ generateSrcFromLf thisPkgId pkgMap m = noLoc mod
     convType =
         \case
             LF.TVar tyVarName ->
-                HsTyVar NoExt NotPromoted $
-                mkRdrName $ LF.unTypeVarName tyVarName
-            LF.TCon LF.Qualified {..} ->
-                HsTyVar NoExt NotPromoted $
-                noLoc $
-                mkOrig
-                    (mkModule
-                         (getUnitId qualPackage)
-                         (mkModuleName $
-                          T.unpack $ LF.moduleNameString qualModule))
-                    (mkOccName varName $
-                     T.unpack $ T.intercalate "." $ LF.unTypeConName qualObject)
+                HsTyVar NoExt NotPromoted $ mkRdrName $ LF.unTypeVarName tyVarName
+            LF.TCon LF.Qualified {..}
+                | [name] <- LF.unTypeConName qualObject ->
+                    HsTyVar NoExt NotPromoted $
+                    noLoc $
+                    mkOrig
+                        (mkModule
+                             (getUnitId qualPackage)
+                             (mkModuleName $
+                              T.unpack $ LF.moduleNameString qualModule))
+                        (mkOccName varName $ T.unpack name)
+                | n@[_name0, _name1] <- LF.unTypeConName qualObject ->
+                    let fs =
+                            fromMaybe
+                                (error $
+                                 "Internal error: Could not find generated record type: " <>
+                                 (T.unpack $ T.intercalate "." n)) $
+                            MS.lookup n sumProdRecords
```

Comment:
`findWithDefault` might be your friend:
https://www.stackage.org/haddock/lts-13.28/containers-0.6.0.1/Data-Map-Strict.html#v:findWithDefault

Reply:
changed.